### PR TITLE
Feat/overdue task

### PR DESCRIPTION
**Pull request by Amandixit10**
### Issue:
Added overdue tab to filter out overdue task.

### Dev Tested:
- [* ] Yes

https://github.com/Real-Dev-Squad/website-status/assets/55151241/ab734e90-61d8-4ec9-ac6e-b06c28c09db9

image

### CODE REVIEW

File: .env.staging
Comment (Pratiyushkumar) [inline .env.staging:1]: why is this change required??

File: __mocks__/handlers/tasks.handler.js
Comment (Pratiyushkumar) [inline __mocks__/handlers/tasks.handler.js:1]: why is this file looking into changes if no change is here

File: __tests__/Unit/Components/Tasks/FilterModal.test.tsx [hunk 1]
```diff
@@ -10,9 +10,9 @@ describe('FilterModal', () => {
     test('renders the modal with correct title and buttons', () => {
         render(
             <FilterModal
-                tabs={[Tab.ASSIGNED, Tab.IN_PROGRESS]}
+                tabs={[Tab.ASSIGNED, Tab.IN_PROGRESS, Tab.OVERDUE]}
                 onSelect={mockOnSelect}
-                activeTab={Tab.ASSIGNED}
+                activeTab={Tab.OVERDUE}
```
Comment (Pratiyushkumar) [inline __tests__/Unit/Components/Tasks/FilterModal.test.tsx:15]: why active tab is overdue instead of assigned

File: src/utils/taskQueryParams.ts
Comment (Pratiyushkumar) [inline src/utils/taskQueryParams.ts:1]: why this file is coming into your changes

File: __tests__/Unit/Components/Tasks/FilterModal.test.tsx (hunk 1 above)
Comment (Amandixit10) [reply __tests__/Unit/Components/Tasks/FilterModal.test.tsx:15]: Here i am testing that if overdue tab is clicked , then it should become active.

File: src/utils/taskQueryParams.ts
Comment (Amandixit10) [reply src/utils/taskQueryParams.ts:1]: Here is a line break.

File: __mocks__/handlers/tasks.handler.js
Comment (Amandixit10) [reply __mocks__/handlers/tasks.handler.js:1]: Here is a line break

**Comment by Amandixit10** [comment]
image

File: src/utils/getActiveTab.ts [hunk 2]
```diff
@@ -18,6 +18,8 @@ export const getActiveTab = (section?: string): Tab => {
             return Tab.MERGED;
         case 'completed':
             return Tab.COMPLETED;
+        case 'overdue':
+            return Tab.OVERDUE;
```
Comment (vinit717) [inline src/utils/getActiveTab.ts:22]: Why this line is not covered in tests?

Comment (Amandixit10) [reply src/utils/getActiveTab.ts:22]: Covered now. Thanks for mentioning.

File: __tests__/Unit/utils/getActiveTab.test.ts [hunk 3]
```diff
@@ -14,4 +14,8 @@ describe('Unit | Util | Get Active Tab', () => {
         expect(getActiveTab('in-progress')).toEqual(Tab.IN_PROGRESS);
         expect(getActiveTab('someRandomSection')).toEqual(Tab.ALL);
     });
+
+    test('returns proper value for the overdue tab when active', () => {
+        expect(getActiveTab('overdue')).toEqual(Tab.OVERDUE);
+    });
```
Comment (Pratiyushkumar) [inline __tests__/Unit/utils/getActiveTab.test.ts:20]: are you sure this test is passing without rendering the component?

Comment (Amandixit10) [reply __tests__/Unit/utils/getActiveTab.test.ts:20]: yes, i have written test separately to check the rendering of overdue tab separately.

Comment (Pratiyushkumar) [reply __tests__/Unit/utils/getActiveTab.test.ts:20]: so why is it written here, please remove it then

Comment (Pratiyushkumar) [reply __tests__/Unit/utils/getActiveTab.test.ts:20]: and also dont resolve comment until reviewers confirmation over the comment

**Review by vinit717 — APPROVED** [review]

**Review by Pratiyushkumar — APPROVED** [review]

**Review by Pratiyushkumar — APPROVED** [review]

**Comment by fakhruddinkw** [comment]
when you passed the assignee: , I see if there are no overdue tasks of the said person, all the overdue tasks are returned. Is this the desired behavior?

**Comment by shubhamsigdar1** [comment]
linked the issue ticket in description

**Comment by shubhamsigdar1** [comment]
I just seen video you create all these task in database??

**Comment by shubhamsigdar1** [comment]
> when you passed the assignee: , I see if there are no overdue tasks of the said person, all the overdue tasks are returned. Is this the desired behavior?

+1

**Review by shubhamsigdar1 — APPROVED** [review]
